test/MetaService.spec.ts

```diff
@@ -1,47 +1,120 @@
+    });
+    it('should return false when memory is null', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      meta.memory = null;
+      const actual = meta.hasMemory();
+      expect(actual).toEqual(false);
+    });
+    it('should return false when memory is false', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      meta.memory = false;
+      const actual = meta.hasMemory();
+      expect(actual).toEqual(false);
+    });
+    it('should return false when memory is not an Object', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      meta.memory = [];
+      const actual = meta.hasMemory();
+      expect(actual).toEqual(false);
+    });
+    it('should return false when memory is an empty Object', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      meta.memory = {};
+      const actual = meta.hasMemory();
+      expect(actual).toEqual(false);
+    });
+    it('should return true when memory is a non-empty Object', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      meta.memory = {test: 'test'};
+      const actual = meta.hasMemory();
+      expect(actual).toEqual(true);
+    });
+  });
+
+  describe('function: getSubFields', () => {
+    it('should be defined', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      const actual = meta.getSubFields;
+      expect(actual).toBeDefined();
+    });
+    it('should return array of fields case 1', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      const fields = 'id,name';
+      const res = meta.getSubFields(fields);
+      expect(res[0]).toBe('id');
+      expect(res[1]).toBe('name');
+    });
+    it('should return array of fields case 2', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      const fields = 'id,name,jobOrders(id,title,status)';
+      const res = meta.getSubFields(fields);
+      expect(res[0]).toBe('id');
+      expect(res[1]).toBe('name');
+      expect(res[2]).toBe('jobOrders(id,title,status)');
+    });
+    it('should return array of fields case 3', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      const fields = 'id,name,jobOrders[5]{status=\'closed\'}(id,title)';
+      const res = meta.getSubFields(fields);
+      expect(res[0]).toBe('id');
+      expect(res[1]).toBe('name');
+      expect(res[2]).toBe('jobOrders(id,title)');
+    });
+    it('should return array of fields case 4', () => {
+      const meta: MetaService = new MetaService('Candidate');
+      const fields = 'id, name, jobOrders(id,title), businessSectors[3](name,id){name=\'Insurance\'}, category';
+      const res = meta.getSubFields(fields);
+      expect(res[0]).toBe('id');
+      expect(res[1]).toBe('name');
+      expect(res[2]).toBe('jobOrders(id,title)');
+      expect(res[3]).toBe('businessSectors(name,id)');
+      expect(res[4]).toBe('category');
+    });
```

Comment:
i think you are missing a test that get the fields inside a nested field
ie.  `meta.getSubFields('businessSectors[3](name,id){name=\'Insurance\'}')`
and `meta.getSubFields('placement(candidate(id,name))`

Reply:
We made a change to this, to only target 'INLINE_EMBEDDED' entities. The only entity that is configured that way it defaultEanCode, and the other earn code entities. The fields for these, do not ever have double nested fields, so the current logic should be able to handle that.